llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -7254,6 +7255,60 @@ bool CombinerHelper::matchAddOverflow(MachineInstr &MI, BuildFnTy &MatchInfo) {
   return false;
 }
 
+bool CombinerHelper::matchShuffleToExtractSubvector(MachineInstr &MI,
+                                                    BuildFnTy &MatchInfo) {
+
+  assert(MI.getOpcode() == TargetOpcode::G_SHUFFLE_VECTOR);
+  const Register DstReg = MI.getOperand(0).getReg();
+  const Register Src1Reg = MI.getOperand(1).getReg();
+  ArrayRef<int> Mask = MI.getOperand(3).getShuffleMask();
+
+  const LLT DstTy = MRI.getType(DstReg);
+  const LLT Src1Ty = MRI.getType(Src1Reg);
+
+  if (!DstTy.isVector() || !Src1Ty.isVector())
+    return false;
```

Comment:
Is this needed? `G_SHUFFLE_VECTOR` only operates on vectors.

Reply:
I think that case could even be matched to a `G_EXTRACT_VECTOR_ELT`. It might not be very useful for AIE's API since element extracts would be implemented using `extractelement` instead of `shufflevector`.

Reply:
Yeah, it is a bit obscure and the overhead it would cause is minimal.